solidity/contracts/OngoingAirdrops.sol

```diff
@@ -91,22 +91,46 @@ contract OngoingAirdrops is AccessControl, IOngoingAirdrops {
     address _claimee,
     TokenAmount[] calldata _tokensAmounts,
     bytes32[] calldata _proof
-  ) external override {}
+  ) external {}
 
   /// @inheritdoc IOngoingAirdrops
   function claimAndTransfer(
     bytes32 _campaign,
     TokenAmount[] calldata _tokensAmounts,
     address _recipient,
     bytes32[] calldata _proof
-  ) external override {}
+  ) external {}
 
   /// @inheritdoc IOngoingAirdrops
   function shutdown(
     bytes32 _campaign,
     IERC20[] calldata _tokens,
     address _recipient
-  ) external override returns (uint256[] memory _unclaimed) {}
+  ) external onlyRole(ADMIN_ROLE) returns (uint256[] memory _unclaimed) {
+    _unclaimed = new uint256[](_tokens.length);
+    // We delete campaign setting it effectively to zero root, so users can't claim this campaign
+    delete roots[_campaign];
+    for (uint256 _i = 0; _i < _tokens.length; ) {
+      // Move from calldata to memory as an optimization
+      IERC20 _token = _tokens[_i];
+      // Build our unique ID for campaign and token address.
+      bytes32 _campaignAndTokenId = _getIdOfCampaignAndToken(_campaign, _token);
+      // Move var from storage to memory
+      uint256 _totalAirdroppedByCampaignAndToken = totalAirdroppedByCampaignAndToken[_campaignAndTokenId];
+      // Understand how much is still available
+      _unclaimed[_i] = _totalAirdroppedByCampaignAndToken - totalClaimedByCampaignAndToken[_campaignAndTokenId];
+      // We update storage so if we call shutdown again we don't break token balances
```

Comment:
Is this the best way to do this? What if we set `totalAirdroppedByCampaignAndToken` and `totalClaimedByCampaignAndToken` to zero? It would be cheaper, and I'm not sure if there is a need to keep any values there